### PR TITLE
Improve hdfs version test to parse CDH version

### PR DESCRIPTION
**Pull request by jiacheliu3**

### CODE REVIEW

File: integration/tools/validation/src/main/java/alluxio/cli/hdfs/HdfsVersionValidationTask.java [hunk 1]
```diff
@@ -31,6 +31,20 @@
...
 
+  // An example output from "hadoop version" command:
+  //    Hadoop 2.7.2
+  //    Subversion https://git-wip-us.apache.org/repos/asf/hadoop.git
+  //      -r b165c4fe8a74265c792ce23f546c64604acf0e41
+  //    Compiled by jenkins on 2016-01-26T00:08Z
+  //    Compiled with protoc 2.5.0
+  //    From source with checksum d0fda26633fa762bff87ec759ebe689c
+  //    This command was run using /tmp/hadoop/share/hadoop/common/hadoop-common-2.7.2.jar
+  private static Pattern sHadoopPattern =
```
Comment (madanadit) [inline integration/tools/validation/src/main/java/alluxio/cli/hdfs/HdfsVersionValidationTask.java:42]: nit: why not `final`?

File: integration/tools/validation/src/main/java/alluxio/cli/hdfs/HdfsVersionValidationTask.java [hunk 2]
```diff
@@ -31,6 +31,20 @@
...
+          Pattern.compile("Hadoop\\s+(?<version>([0-9]\\.)+[0-9]+)");
+  // An example Hadoop version for CDH distribution is
+  // Hadoop 2.6.0-cdh5.16.2
+  private static Pattern sCdhPattern = Pattern.compile("cdh(?<cdhVersion>([0-9]+\\.)+[0-9]+)");
```
Comment (jiacheliu3) [inline integration/tools/validation/src/main/java/alluxio/cli/hdfs/HdfsVersionValidationTask.java:46]: @madanadit Do we want to consider HDP?

File: integration/tools/validation/src/main/java/alluxio/cli/hdfs/HdfsVersionValidationTask.java [hunk 3]
```diff
@@ -39,11 +39,12 @@
   //    Compiled with protoc 2.5.0
   //    From source with checksum d0fda26633fa762bff87ec759ebe689c
   //    This command was run using /tmp/hadoop/share/hadoop/common/hadoop-common-2.7.2.jar
-  private static Pattern sHadoopPattern =
+  private final static Pattern sHadoopPattern =
```
Comment (madanadit) [inline integration/tools/validation/src/main/java/alluxio/cli/hdfs/HdfsVersionValidationTask.java:42]: @jiacheliu3 same for `sCdhPattern` and change capitalization to `HADOOP_PATTERN`

File: integration/tools/validation/src/main/java/alluxio/cli/hdfs/HdfsVersionValidationTask.java (hunk 2 above)
Comment (madanadit) [reply integration/tools/validation/src/main/java/alluxio/cli/hdfs/HdfsVersionValidationTask.java]: nope

**Comment by madanadit** [comment]
alluxio-bot, merge this please.

**Comment by madanadit** [comment]
@ZacBlanco need to cherry-pick this to branch-2.3.

**Comment by ZacBlanco** [comment]
+1

**Comment by madanadit** [comment]
alluxio-bot, cherry-pick this to branch-2.3 please.

**Pull request merged**